packages/core/content-manager/server/src/history/controllers/history-version.ts

```diff
@@ -18,21 +18,22 @@ const createHistoryVersionController = ({ strapi }: { strapi: Strapi }) => {
         throw new errors.ForbiddenError('contentType and documentId are required');
       }
 
-      const params = ctx.query as HistoryVersions.GetHistoryVersions.Request['query'];
-
       /**
        * There are no permissions specifically for history versions,
        * but we need to check that the user can read the content type
        */
       const permissionChecker = getContentManagerService('permission-checker').create({
         userAbility: ctx.state.userAbility,
-        model: params.contentType,
+        model: ctx.query.contentType,
       });
 
       if (permissionChecker.cannot.read()) {
         return ctx.forbidden();
       }
 
+      const params: HistoryVersions.GetHistoryVersions.Request['query'] =
+        await permissionChecker.sanitizeQuery(ctx.query);
+
       const { results, pagination } = await getService(strapi, 'history').findVersionsPage(params);
 
       return { data: results, meta: { pagination } };
```

Comment:
are results of the history version sanitized? Do user permissions apply to the specific versions of documents being returned?

Reply:
We already do a bit of sanitization [inside](https://github.com/strapi/strapi/blob/82191cda0e69ccbe243ab1d7d8ea57e70309a7ad/packages/core/content-manager/server/src/history/services/history.ts#L168) of the service, but not much. We'll need to do more in the future, like removing passwords, and checking field-level permissions ideally (that check is only in the frontend currently)

The tricky thing we can't just use sanitizeOutput from the permission checker. Because the sanitization on each history version needs to be done against the schema of that specific history version (it can change over time with CTB changes). Also we want to keep the hidden fields, in order to show a true history of the data.